### PR TITLE
Show documentation on signature help

### LINKED ISSUE #420: Show builtin API docs as overlay documentation in VSCode
The markdown extracted from documentation comments in #419 should be shown in an overlay when operator names are being hovered.

### PR DESCRIPTION
**Pull request by bugarela**
Hello :octocat: 

This helps VSCode user to quickly find documentation for operators they are trying to use, by showing them our API reference documentation for builtin operators. It will also show user-defined docstrings for user-defined operators, as long as they are in the same module (for now).

The documentation for the builtin definitions is fetched from `builtin.qnt`, which I moved to `src/` and it's now being copied to `dist/` on the `npm run compile` script.

I'll open aditional PRs in the near future to extend this feature, by
1. Also showing the documentation when hovering over a name
2. Showing the inferred types for non-annotated definitions

image

Closes #420 

- [X] Tests added for any new code
- [X] Ran `npm run format` (or had formatting run automatically on all files edited)
- [X] Documentation added for any new functionality
- [x] Entries added to the respective `CHANGELOG.md` for any new functionality

### CODE REVIEW

File: quint/src/builtin.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { ErrorMessage, parsePhase1 } from "./quintParserFrontend";
+import { Either } from "@sweet-monads/either";
+import { lf } from "eol";
+
+/**
+ * The documentation map for the builtin definitions
+ *
+ * @returns a map of builtin definition names to their documentation
+ */
+export function builtInDocs(): Either<ErrorMessage[], Map<string, DocumentationEntry>> {
```
Comment (shonfeder) [inline quint/src/builtin.ts:27]: it looks like you treat this as one word everywhere else:

```suggestion
export function builtinDocs(): Either<ErrorMessage[], Map<string, DocumentationEntry>> {
```

File: quint/src/builtin.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const path = resolve(__dirname, 'builtin.qnt')
+  const sourceCode = lf(readFileSync(path, 'utf8'))
```
Comment (shonfeder) [inline quint/src/builtin.ts:29]: a comment to note what lf is used for might help

File: quint/src/docs.ts [hunk 3]
```diff
@@ -1,20 +1,62 @@
...
+ * Produces a documentation map for the modules' definitions
+ *
+ * @param quintModule the module for which documentation should be produced
+ * @returns a map of definition names to their documentation
+ */
+export function produceDocs(quintModule: QuintModule): Map<string, DocumentationEntry> {
+  const entries = quintModule.defs.map((def) => {
+    if (def.kind === 'module') {
+      // TODO: Produce documentation for nested modules
+      return undefined
```
Comment (shonfeder) [inline quint/src/docs.ts:40]: Why return undefined here rather than an empty map?

Comment (shonfeder) [reply quint/src/docs.ts:40]: Ah, I think it relates to the use of compact from lodash? A comment explaining would have helped me here.

File: quint/test/docs.test.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { assert } from 'chai'
+import { describe, it } from 'mocha'
+import { produceDocs, toMarkdown } from '../src/docs'
+import { buildModuleWithDefs } from './builders/ir'
+
+describe('produceDocs', () => {
+  const module = buildModuleWithDefs([
+    `/// This is a docstring for foo\nval foo = 1`,
```
Comment (shonfeder) [inline quint/test/docs.test.ts:8]: Could this and similar be multi line strings? I think it would help with readability.

**Review by shonfeder — APPROVED** [review]
All looks great to me!

I left a few suggestions but nothing important. Screenshot looks wonderful!

**Comment by konnov** [comment]
This is amazing!

**Pull request merged**